client/my-sites/email/titan-redirector/titan-redirector.jsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import page from 'page';
+import { map, fromPairs } from 'lodash-es';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'calypso/lib/wp';
+import EmptyContent from 'calypso/components/empty-content';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getManagePurchaseUrlFor } from 'calypso/my-sites/purchases/paths';
+import { emailManagementNewTitanAccount } from 'calypso/my-sites/email/paths';
+import getSitesItems from 'calypso/state/selectors/get-sites-items';
+import { getSiteSlug, hasAllSitesList } from 'calypso/state/sites/selectors';
+import { SUPPORT_ROOT } from 'calypso/lib/url/support';
+import { addQueryArgs } from 'calypso/lib/route';
+import { login } from 'calypso/lib/paths';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import QuerySites from 'calypso/components/data/query-sites';
+
+class TitanRedirector extends Component {
+	state = {
+		loaded: false,
+		hasError: false,
+		action: null,
+		orderId: null,
+		subscriptionId: null,
+		blogId: null,
+		domain: null,
+		error: null,
+	};
+
+	componentDidMount() {
+		const { mode, jwt, isLoggedIn } = this.props;
+
+		if ( ! isLoggedIn ) {
+			return;
+		}
+
+		wp.undocumented()
+			.getTitanDetailsForIncomingRedirect( mode, jwt )
+			.then(
+				( data ) => {
+					this.setState( {
+						hasError: false,
+						loaded: true,
+						action: data?.action,
+						orderId: data?.order_id,
```

Comment:
Seems like the `orderId` is never used? Are we planning to use it? Guess we can remove it if it won't immediately be used.